### PR TITLE
Basic file serving

### PR DESCRIPTION
**Pull request by ddaze**

### CODE REVIEW

File: invenio_files_rest/views.py [hunk 1]
```diff
@@ -157,10 +163,11 @@ def post(self, **kwargs):
         args = parser.parse(self.post_args, request)
         try:
             if args['location_id']:
-                location = Location.get(args['location_id'])
+                # TODO: Check why query is used directly.
+                location = Location.query.get(args['location_id'])
```
Comment (lnielsen) [inline invenio_files_rest/views.py]: ``Location.get_by_name()``

File: invenio_files_rest/views.py [hunk 2]
```diff
@@ -157,10 +163,11 @@ def post(self, **kwargs):
...
             else:
                 # Get one of the active locations
-                location = Location.all().first()
+                location = Location.first()
```
Comment (lnielsen) [inline invenio_files_rest/views.py]: ``Location.get_default()``

File: invenio_files_rest/models.py [hunk 3]
```diff
@@ -109,6 +109,11 @@ def all(cls):
         """Return query that fetches all locations."""
         return Location.query.all()
 
+    @classmethod
```
Comment (lnielsen) [inline invenio_files_rest/models.py]: See comment in ``views.py``.

File: invenio_files_rest/views.py (hunk 1 above)
Comment (ddaze) [inline invenio_files_rest/views.py]: Should I also rename args['location_id'] to 'location_name' or 'location'?

**Comment by drjova** [comment]
@lnielsen is it ready?

**Comment by lnielsen** [comment]
@drjova You can merge. Running to a meeting now :-)

**Comment by drjova** [comment]
@ddaze could you please rebase :)

**Comment by egabancho** [comment]
I think you'll have to rebase again :relaxed:

**Comment by ddaze** [comment]
okay :walking:  :D

File: tests/test_views.py [hunk 4]
```diff
@@ -27,38 +27,39 @@
...
-#     """Test post a bucket."""
-#     with app.test_client() as client:
-#         resp = client.post(
-#             '/files',
-#             headers={'Content-Type': 'application/json', 'Accept': '*/*'}
-#         )
-#         assert resp.status_code == 200
-#         data = json.loads(resp.data)
-#         assert 'url' in data
+def test_get_buckets(app, dummy_location):
```
Comment (egabancho) [inline tests/test_views.py]: Do you really need the `dummy_location` here?

File: tests/test_views.py [hunk 5]
```diff
@@ -27,38 +27,39 @@
...
+        # With location_name
+        resp = client.post(
+            '/files',
+            data=json.dumps({'location_name': dummy_location.name}),
+            headers={'Content-Type': 'application/json', 'Accept': '*/*'}
+        )
+        assert resp.status_code == 200
+
+
+def test_post_bucket(app, dummy_location):
```
Comment (egabancho) [inline tests/test_views.py]: Do you really need the `dummy_location` here?

Comment (ddaze) [inline tests/test_views.py]: Yes, without a location no bucket can be created.

Comment (egabancho) [inline tests/test_views.py]: :+1:

**Pull request merged**